File: java/p_0001.java
/*
 * Find the sum of all the multiples of 3 or 5 below 1000.
 */

// import java.util.stream.IntStream;

public class p_0001 {
    public static void main(String [] args) {
        int sum = 0;
        for (int i = 1; i < 1000; i++) {
            if ((i%3 == 0) || (i%5 == 0)) {
                sum += i;
            }
        }
        // long sum = IntStream.range(0, 1000).filter(item -> (item%3 == 0) || (item%5 == 0)).reduce(0, (a,b)->a+b);
        System.out.printf("Sum is %d\n", sum);
    }
}

File: cpp/p_0001.cc
/* Find the sum of all the multiples of 3 and 5 below 1000 */

#include <iostream>
#include <numeric>
#include <ranges>

auto p_0001(int max) {
    auto numbers = std::ranges::iota_view{1, max};
    return std::accumulate(numbers.begin(), numbers.end(), 0,
                           [](int sum, int x) {
                               if ((x % 3 == 0) || (x % 5 == 0)) {
                                   sum += x;
                               }
                               return sum;
                           });
}

int main(void) {
    std::cout << "Problem 0001: Sum of multiples of 3 and 5 below 1000 is: "
              << p_0001(1000) << '\n';
}
